### PR TITLE
replace yarn with pnpm

### PR DESCRIPTION
**Pull request by m1m1sha**
replace yarn with pnpm

### CODE REVIEW

**Comment by m1m1sha** [comment]
in the past, i used `tauri/action` to build, so i am not very clear about what happened

checked and found the following errors

### aarch64-unknown-linux-gnu error: 

````
gui_target: aarch64-unknown-linux-gnu

# cmd
pnpm tauri build -- --target $GUI_TARGET --verbose --bundles deb

# error message
error: unexpected argument '--bundles' found

````

### archive artifact error: 
````
target: mips-unknown-linux-musl

````

### other error is can not find the release file: 

````
gui_target: aarch64-unknown-linux-gnu | x86_64-unknown-linux-gnu | x86_64-apple-darwin | aarch64-apple-darwin | x86_64-pc-windows-msvc

# cmd
pnpm tauri build -- --target $GUI_TARGET --verbose

# error message
No such file or directory (os error 2)

````


if available, please provide assistance to resolve these errors

**Comment by KKRainbow** [comment]
okay, I'll take a look at night.

**Comment by m1m1sha** [comment]
i seem to have solved this problem by using `tauri/action`
https://github.com/m1m1sha/EasyTier/actions/runs/8969651756/job/24631460890

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -49,103 +51,131 @@ jobs:
...
+
+      - name: Setup pnpm cache
+        uses: actions/cache@v4
+        with:
+          path: ${{ env.STORE_PATH }}
+          key: ${{ runner.os }}-pnpm-store-${{ hashFiles('**/pnpm-lock.yaml') }}
+          restore-keys: |
+            ${{ runner.os }}-pnpm-store-
+
+      - name: Install frontend dependencies
```
Comment (KKRainbow) [inline .github/workflows/rust.yml:79]: 这一步感觉放在 gui compile 那一步里会好点，如果不用编译 gui，也不用安装 pnpm 的依赖

Comment (m1m1sha) [reply .github/workflows/rust.yml:79]: 的确

**Pull request merged**